phpBB/phpbb/plupload/plupload.php

```diff
@@ -276,22 +276,47 @@ public function generate_resize_string()
 	}
 
 	/**
-	* Checks various php.ini values and the maximum file size to determine
-	* the maximum size chunks a file can be split up into for upload
+	* Checks various php.ini values to determine the maximum chunk
+	* size a file should be split into for upload.
+	*
+	* The intention is to calculate a value which reflects whatever
+	* the most restrictive limit is set to.  And to then set the chunk
+	* size to half that value, to ensure any required transfer overhead
+	* and POST data remains well within the limit.  Or, if all of the
+	* limits are set to unlimited, the chunk size will also be unlimited.
 	*
 	* @return int
 	*/
 	public function get_chunk_size()
 	{
-		$max = min(
-			$this->php_ini->getBytes('upload_max_filesize'),
-			$this->php_ini->getBytes('post_max_size'),
-			max(1, $this->php_ini->getBytes('memory_limit')),
-			$this->config['max_filesize']
-		);
+		$max = 0;
+
+		// unlimited is -1 for memory_limit. 0 should be an invalid configuration.
+		$limit_memory = $this->php_ini->getBytes('memory_limit');
+
+		if ($limit_memory > 0)
```

Comment:
If not null is returned by `php_ini->getBytes()` it is safe to assume that true is more than 0.

`if ($limit_memory)`

Applies everywhere here.

Reply:
Yeah,
`if (true === $limit_memory)`, nitpicking though.

Agreed that negative values are always `true` in PHP.